src/session.rs

```diff
@@ -173,22 +173,33 @@ impl Ongoing {
         Poll::Ready(rt)
     }
 }
+pub(crate) trait Connector:
```

Comment:
Let's stick to just giving the bounds directly. Intermediate traits tend to cause pain with type inference. One day we'll have proper trait aliases, but for now let's repeat the bound :)

Reply:
Yeah, probably :), Shame there are no trait aliases yet. I'll change it to normal bounds